src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,77 @@
+package duke;
+
+import duke.task.*;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class Ui {
+    private static final String LINE = "     __________________________________________________\n"; // 5 spaces, 50 dashes
+    private static final String INDENT = "     "; // 5 spaces
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    public static void printWelcome() {
+        String logo = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+
+        String greeting = LINE
+            + INDENT + "Hello! I'm Duke\n"
+            + INDENT + "What can I do for you?\n"
+            + LINE;
+        System.out.println(greeting);
+    }
+
+    public static String readInput() {
+        return SCANNER.nextLine();
+    }
+
+    public static void printList(List<Task> listToPrint) {
+        System.out.println(LINE + INDENT + "Here are the tasks in your list:");
+        for (int i = 0; i < listToPrint.size(); i++) {
+            Task currTask = listToPrint.get(i);
+            System.out.println(INDENT + (i + 1) + "." + currTask.toString());
+        }
+        System.out.println(LINE);
+    }
+
+    public static void printError(String message) {
+        System.out.println(LINE + INDENT + message + LINE);
+    }
+
+    public static void printDoneTask(String taskString) {
+        System.out.println(LINE + INDENT + "Nice! I've marked this task as done:\n"
+            + INDENT + INDENT + taskString + "\n"
+            + LINE);
```

Comment:
Perhaps wrapped lines could be indented 8 spaces (instead of 4) more than the parent line.